src/modules/frost/keygen_impl.h

```diff
@@ -337,7 +387,63 @@ int secp256k1_frost_share_verify(const secp256k1_context* ctx, size_t threshold,
         return 0;
     }
 
-    return secp256k1_frost_vss_verify_internal(ctx, threshold, id33, &share_i, vss_commitment);
+    return secp256k1_frost_vss_verify_internal(ctx, threshold, id33, &share_i, vss_commitment, 0);
+}
+
+int secp256k1_frost_refresh_share(const secp256k1_context *ctx, secp256k1_frost_share *share_after_refresh, secp256k1_pubkey * *vss_commitments_after_refresh, const secp256k1_frost_share *share, const secp256k1_pubkey * const *vss_commitments, const secp256k1_frost_share * const *refresh_shares, const secp256k1_pubkey * const *refresh_vss_commitments, size_t threshold, size_t n_participants, const unsigned char *id33) {
+    secp256k1_scalar acc;
+    size_t i, j;
+    int ret = 1;
+
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(share_after_refresh != NULL);
+    memset(share_after_refresh, 0, sizeof(secp256k1_frost_share));
+    ARG_CHECK(vss_commitments_after_refresh != NULL);
+    ARG_CHECK(share != NULL);
+    ARG_CHECK(vss_commitments != NULL);
+    ARG_CHECK(refresh_shares != NULL);
+    ARG_CHECK(refresh_vss_commitments != NULL);
+    ARG_CHECK(id33 != NULL);
+    ARG_CHECK(n_participants > 1);
+    ARG_CHECK(threshold > 1);
+
+    if (threshold > n_participants) {
+        return 0;
+    }
+
+    if (!secp256k1_frost_share_load(ctx, &acc, share)) {
+        return 0;
+    }
+    for (i = 0; i < n_participants; i++) {
+        secp256k1_scalar refresh_share_i;
+        if (!secp256k1_frost_share_load(ctx, &refresh_share_i, refresh_shares[i])) {
+            return 0;
+        }
+
+        /* Verify each refresh share against its commitments. */
+        ret &= secp256k1_frost_vss_verify_internal(ctx, threshold, id33, &refresh_share_i, &refresh_vss_commitments[i], 1);
+
+        secp256k1_scalar_add(&acc, &acc, &refresh_share_i);
+        for (j = 0; j < threshold; j++) {
```

Comment:
We can perform the point operations more efficiently with `secp256k1_ecmult_multi_var`.

Reply:
Never mind, chatted with @wpaulino, `secp256k1_ecmult_multi_var` can't be used here because we are doing pairwise addition, not summing the entire vector.